src/UI/WB.UI.Designer/Controllers/Api/WebTester/ScenariosController.cs

```diff
@@ -75,7 +77,16 @@ public async Task<IActionResult> Get(string token, int scenarioId)
             if (scenario == null)
                 return NotFound(new {Message = "Scenario not found"});
             if (questionnaire != scenario.QuestionnaireId)
-                return Forbid("Scenario from other questionnaire");
+            {
+                var anonymousQuestionnaire = this.dbContext.AnonymousQuestionnaires

```

Comment:
Since AnonymousQuestionnaireId is primary key, and the object is small, and we do not have compound index maybe it's better to get a record and check it in the code.

Reply:
Discussed